packages/http/src/shared.ts

```diff
@@ -5,31 +5,211 @@ export type TActivityResponse = definitions["v1ActivityResponse"];
+  definitions["v1SignTransactionResult"]["signedTransaction"];
 
 export class TurnkeyActivityError extends Error {
-  activityId: TActivityId | null;
-  activityStatus: TActivityStatus | null;
-  activityType: TActivityType | null;
-  cause: Error | null;
+  activityId: TActivityId | undefined;
+  activityStatus: TActivityStatus | undefined;
+  activityType: TActivityType | undefined;
+  cause: Error | undefined;
 
   constructor(input: {
     message: string;
-    cause?: Error | null;
-    activityId?: TActivityId | null;
-    activityStatus?: TActivityStatus | null;
-    activityType?: TActivityType | null;
+    cause?: Error | undefined;
+    activityId?: TActivityId | undefined;
+    activityStatus?: TActivityStatus | undefined;
+    activityType?: TActivityType | undefined;
   }) {
     const { message, cause, activityId, activityStatus, activityType } = input;
     super(message);
 
     this.name = "TurnkeyActivityError";
-    this.activityId = activityId ?? null;
-    this.activityStatus = activityStatus ?? null;
-    this.activityType = activityType ?? null;
-    this.cause = cause ?? null;
+    this.activityId = activityId ?? undefined;
+    this.activityStatus = activityStatus ?? undefined;
+    this.activityType = activityType ?? undefined;
+    this.cause = cause ?? undefined;
   }
 }
 
+export class TurnkeyActivityConsensusNeededError extends Error {
+  activityId: TActivityId | undefined;
+  activityStatus: TActivityStatus | undefined;
+  activityType: TActivityType | undefined;
+  cause: Error | undefined;
+
+  constructor(input: {
+    message: string;
+    cause?: Error | undefined;
+    activityId?: TActivityId | undefined;
+    activityStatus?: TActivityStatus | undefined;
+    activityType?: TActivityType | undefined;
+  }) {
+    const { message, cause, activityId, activityStatus, activityType } = input;
+    super(message);
+
+    this.name = "TurnkeyActivityConsensusNeededError";
+    this.activityId = activityId ?? undefined;
+    this.activityStatus = activityStatus ?? undefined;
+    this.activityType = activityType ?? undefined;
+    this.cause = cause ?? undefined;
+  }
+}
+
+export class InvalidArgumentError extends Error {
+  activityId: TActivityId | undefined;
+  activityStatus: TActivityStatus | undefined;
+  activityType: TActivityType | undefined;
+  cause: Error | undefined;
+
+  constructor(input: {
+    message: string;
+    cause?: Error | undefined;
+    activityId?: TActivityId | undefined;
+    activityStatus?: TActivityStatus | undefined;
+    activityType?: TActivityType | undefined;
+  }) {
+    const { message, cause, activityId, activityStatus, activityType } = input;
+    super(message);
+
+    this.name = "InvalidArgumentError";
+    this.activityId = activityId ?? undefined;
+    this.activityStatus = activityStatus ?? undefined;
+    this.activityType = activityType ?? undefined;
+    this.cause = cause ?? undefined;
+  }
+}
+
+export function assertActivityCompleted(input: {
```

Comment:
Now that I'm seeing this with fresh eyes: the activity ID isn't used anywhere really, we're simply comparing the passed in `status` to `CONSENSUS_NEEDED` and `COMPLETED`. Don't think it's necessarily worth changing this but...a bit of an odd interface given nothing guarantees the passed in status come from the passed in activity ID :/

Maybe it'd be clearer if a `TActivity` was expected and passed in instead?

Reply:
Attempting this here: https://github.com/tkhq/sdk/pull/350/commits/aa16d4d5c4e130a8ede3946eca3ec13868ed437d

The drawback of this approach is that it's a bit duplicative -- e.g. we return something like (paraphrasing)

{
  ...
  result: activityData["result"]["activityResult"],
  activity: activityData
  ...
}

So you see the overlap there, but having the `result` up front and center just makes access easier. The reason we needed this in the first place was originally for type-compatibility, but I realized it generally could be helpful to return all activity details

Reply:
yeah it's definitely not super clear-cut, but I like this new version: `assertActivityCompleted(activity);` reads suuuper nice 😍